src/libs/OptionsListUtils.js

```diff
@@ -179,7 +179,9 @@ function getOptions(reports, personalDetails, draftComments, activeReportID, {
         // Skip this entry if it has no comments and is not the active report. We will only show reports from
         // people we have sent or received at least one message with.
         const hasNoComments = report.lastMessageTimestamp === 0;
-        if (!showReportsWithNoComments && hasNoComments && report.reportID !== activeReportID) {
+        const shouldFilterReport = !showReportsWithNoComments && hasNoComments && report.reportID !== activeReportID
+            && !report.isPinned;
```

Comment:
This PR looks good overall, just have one minor request: Having this condition by itself on a separate line looks odd. Either put them all on the same line (I think the linter just warns about being over 120 chars, which is fine) or put each condition on its own line.